### ISSUE TITLE
design-this

### ISSUE THREAD

**Issue by adamcharlton** [adamcharlton/AdventureMoney#1]
Design this AdventureMoney app.

Built on Liberty with IntelliJ and Gradle and deployed locally and to BlueMix.

Fairly sure this is the old app I used to use and love - https://code.google.com/p/adventure-money/
Want to model on that as it's familiar.

@sjhogan this is seriously too easy to get running, had the liberty app up bluemix with intellij in like 15 mins.

**Comment by sjhogan** [comment]
@adamcharlton I'll try to get some time today to look at the original. Could probably give you a hand with development of the UI if you want the assistance.

**Comment by adamcharlton** [comment]
Yeah it's something I want to work on slowly to keep my Liberty codefu sharp. 

I found this last night which had some screenshots and a list of the Python dependencies. I'm also using the Python plugin from pycharm in IntelliJ to run it. 

http://laszlopandy.com/2007/05/13/presenting-adventure-money/

**Comment by adamcharlton** [comment]
@sjhogan So had a bit of a look into Node frameworks and there's a few that look nice.

http://hapijs.com/ is definately an option.
Also http://koajs.com/ and http://sailsjs.org/ (this one is NodeJS MVC)

Hapi looks good enough to start with, I can play with Sails next.

**Comment by sjhogan** [comment]
All worthy choices. Sails builds on top of Express & is sort of a full stack solution a la Rails. Koa's from the same author as Express, but is pretty different. Hapi doesn't build on Express & purposefully emphasises configuration (I think the intention is to make the code base more self documenting for ease of maintenance etc). Express is almost a de facto standard for node web apps, so it would be interesting to check out the pros & cons of something different.

You might also want to look at Kraken.js, it's more of a set of conventions built on Express rather than a separate framework. Came out of PayPal.
